[feature request] Ignore's Django model's Meta

I'm working in a code base that uses "model mixins" extensively:

```python
from django.db import models

class ArchivableModel(models.Model):
    class Meta:
        abstract = True
    archived = models.BooleanField(default=False)

class Tracked(models.Model):
    class Meta:
        abstract = True
    created_at = models.DateTimeField(auto_now=True)

class Model(Tracked, ArchivableModel):
    pass
```

`zuban check` rejects this with:

```bash
$ uvx --with django  zuban check metas.py
metas.py:17: error: Definition of "Meta" in base class "Tracked" is incompatible with definition in base class "ArchivableModel"  [misc]
Found 1 error in 1 file (checked 1 source file)
```

`mypy` (on its own) would raise the same issue, but with the help of django-stubs extension it doesn't:

```
[mypy]
plugins =
    mypy_django_plugin.main

[mypy.plugins.django-stubs]
django_settings_module = "settings"   # empty file
```

```bash
$ uvx --with django --with "django-stubs[compatible-mypy]" mypy metas.py
Success: no issues found in 1 source file
```

--- a/crates/zuban_python/src/file/diagnostics.rs
+++ b/crates/zuban_python/src/file/diagnostics.rs
@@ -1097,10 +1097,18 @@ impl Inference<'_, '_, '_> {
             self.i_s,
             || class_infos.base_types(),
             // Don't check symbols if they are part of the instance that we are currently using.
+            // Also, Django models routinely combine abstract "model mixins" that each declare their
+            // own inner "Meta" class. Those inner classes are unrelated and would otherwise be
+            // reported as incompatible, but the django-stubs mypy plugin ignores them, so we do the
+            // same for any class derived from the django-stubs base classes.
             |name| {
-                c.lookup_symbol(self.i_s, name)
-                    .into_maybe_inferred()
-                    .is_none()
+                if name == "Meta" && c.has_django_stubs_base_class(self.i_s.db) {
+                    return false;
+                } else {
+                    c.lookup_symbol(self.i_s, name)
+                        .into_maybe_inferred()
+                        .is_none()
+                }
             },
             |issue| c.add_issue_on_args(self.i_s, issue),
         );

--- a/crates/zuban_python/tests/mypylike/tests/django.test
+++ b/crates/zuban_python/tests/mypylike/tests/django.test
@@ -56,3 +56,22 @@ def f(album: Album):
 __main__.py:25:complete -> [artist, artist_id]
 __main__.py:27:complete -> [composer, composer_id]
 __main__.py:29:complete -> [name]
+
+[case django_abstract_model_mixins_meta]
+# From GH #470
+from django.db import models
+
+class ArchivableModel(models.Model):
+    class Meta:
+        abstract = True
+    archived = models.BooleanField(default=False)
+
+class Tracked(models.Model):
+    class Meta:
+        abstract = True
+    created_at = models.DateTimeField(auto_now=True)
+
+# The incompatible "Meta" inner classes of the bases must not be reported for
+# Django models, mirroring the behaviour of the django-stubs mypy plugin.
+class Model(Tracked, ArchivableModel):
+    pass

